magicctapipe/io/io.py

```diff
@@ -619,19 +619,30 @@ def load_mc_dl2_data_file(input_file, quality_cuts, irf_type, dl2_weight_type):
         * len(np.unique(event_table["obs_id"]))
     )
 
+    min_viewcone_radius = sim_config["min_viewcone_radius"][0] * u.deg
+    max_viewcone_radius = sim_config["max_viewcone_radius"][0] * u.deg
+
+    viewcone_diff = max_viewcone_radius - min_viewcone_radius
+
+    if viewcone_diff < u.Quantity(0.001, u.deg):
+        # Handle ring-wobble MCs as same as point-like MCs
+        viewcone = 0 * u.deg
+    else:
+        viewcone = max_viewcone_radius
```

Comment:
I see that `SimulationConfigContainer` has a `diffuse` attribute. Couldn't we use that to tell if it is ring wobble/point-like or diffuse and assign the `viewcone` value accordingly?

Reply:
Thanks, I checked a ring-wobble MC data, and the `diffuse` attribute is flagged as 1. I also checked that my point MC is `diffuse = 0`, and the all-sky training diffuse MC is `diffuse = 1`, so it cannot be used for the separation.